CLI SIGSEGV converting channel from private to public

#### Summary
Converting a channel from private to public via the CLI produces a SIGSEGV and succeeds.

#### Steps to reproduce
Server version 5.16.2, Database schema 5.16.0 hosted on Postgres
Host is Ubuntu 18.04.3 LTS.

ssh to server, and from the mattermost installation directory:

`sudo ./bin/mattermost channel modify my_team:existing_channel --username chris.herborth --public`

#### Expected behavior
I expect the channel to go from private to public.

#### Observed behavior (that appears unintentional) 
The channel went from private to public, but the CLI also produced this error:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xfacf02]

goroutine 1 [running]:
github.com/mattermost/mattermost-server/app.(*App).postChannelPrivacyMessage(0xc003a0b880, 0x0, 0xc0026ba4e0, 0x0)
	/go/src/github.com/mattermost/mattermost-server/app/channel.go:584 +0x2b2
github.com/mattermost/mattermost-server/app.(*App).UpdateChannelPrivacy(0xc003a0b880, 0xc0026ba4e0, 0x0, 0x0, 0x0)
	/go/src/github.com/mattermost/mattermost-server/app/channel.go:555 +0x6c
github.com/mattermost/mattermost-server/cmd/mattermost/commands.modifyChannelCmdF(0x29df060, 0xc000268740, 0x1, 0x4, 0x0, 0x0)
	/go/src/github.com/mattermost/mattermost-server/cmd/mattermost/commands/channel.go:522 +0x31a
github.com/spf13/cobra.(*Command).execute(0x29df060, 0xc0002686c0, 0x4, 0x4, 0x29df060, 0xc0002686c0)
	/go/src/github.com/mattermost/mattermost-server/vendor/github.com/spf13/cobra/command.go:826 +0x465
github.com/spf13/cobra.(*Command).ExecuteC(0x29d7360, 0x14d8715, 0xc0001b9f88, 0xc00009a058)
	/go/src/github.com/mattermost/mattermost-server/vendor/github.com/spf13/cobra/command.go:914 +0x2fc
github.com/spf13/cobra.(*Command).Execute(...)
	/go/src/github.com/mattermost/mattermost-server/vendor/github.com/spf13/cobra/command.go:864
github.com/mattermost/mattermost-server/cmd/mattermost/commands.Run(...)
	/go/src/github.com/mattermost/mattermost-server/cmd/mattermost/commands/root.go:15
main.main()
	/go/src/github.com/mattermost/mattermost-server/cmd/mattermost/main.go:30 +0x89
```

#### Possible fixes
Sorry, I'm not much of a Go dev.

--- a/cmd/mattermost/commands/channel.go
+++ b/cmd/mattermost/commands/channel.go
@@ -519,6 +519,10 @@ func modifyChannelCmdF(command *cobra.Command, args []string) error {
 	}
 
 	user := getUserFromUserArg(a, username)
+	if user == nil {
+		return fmt.Errorf("Unable to find user: '%v'", username)
+	}
+
 	if _, err := a.UpdateChannelPrivacy(channel, user); err != nil {
 		return errors.Wrapf(err, "Failed to update channel ('%s') privacy", args[0])
 	}

--- a/cmd/mattermost/commands/channel_test.go
+++ b/cmd/mattermost/commands/channel_test.go
@@ -227,3 +227,30 @@ func Test_searchChannelCmdF(t *testing.T) {
 		})
 	}
 }
+
+func TestModifyChannel(t *testing.T) {
+	th := Setup().InitBasic()
+	defer th.TearDown()
+
+	channel1 := th.CreatePrivateChannel()
+	channel2 := th.CreatePrivateChannel()
+
+	th.CheckCommand(t, "channel", "modify", "--public", th.BasicTeam.Name+":"+channel1.Name, "--username", th.BasicUser2.Email)
+	res, err := th.App.Srv.Store.Channel().Get(channel1.Id, false)
+	require.Nil(t, err)
+	assert.Equal(t, model.CHANNEL_OPEN, res.Type)
+
+	// should fail because user doesn't exist
+	require.Error(t, th.RunCommand(t, "channel", "modify", "--public", th.BasicTeam.Name+":"+channel2.Name, "--username", "idonotexist"))
+
+	pchannel1 := th.CreatePublicChannel()
+	pchannel2 := th.CreatePublicChannel()
+
+	th.CheckCommand(t, "channel", "modify", "--private", th.BasicTeam.Name+":"+pchannel1.Name, "--username", th.BasicUser2.Email)
+	res, err = th.App.Srv.Store.Channel().Get(pchannel1.Id, false)
+	require.Nil(t, err)
+	assert.Equal(t, model.CHANNEL_PRIVATE, res.Type)
+
+	// should fail because user doesn't exist
+	require.Error(t, th.RunCommand(t, "channel", "modify", "--private", th.BasicTeam.Name+":"+pchannel2.Name, "--username", "idonotexist"))
+}

